### ISSUE TITLE
copy.copy() should always be replaced by copy.deepcopy()

### ISSUE THREAD

**Issue by pengzhenghao** [metadriverse/metadrive#261]
As title.

**Comment by QuanyiLi** [comment]
For config and some ```class Config```objects, it works. But for other objects like ```class Lane```, we don't like it to work in that way, as there might be some Panda/Bullet objects in the instance variables which can not be deep copied. In this case, we actually want a shallow copy. 

If you run the main branch with ```debug=True```, you will see a lot of error messages saying bullet objects can not be deep copied. I already fixed it in the #310

**Comment by pengzhenghao** [comment]
Forget this issue! The copy problem is largely resolved when I fixed the memory leak. And everything here now are just quite good and don’t need to changeBest,ZhenghaoOn Feb 24, 2023, at 04:18, Quanyi Li ***@***.***> wrote:﻿
For config and some class Configobjects, it works. But for other objects like class Lane, we don't like it to work in that way, as there might be some Panda/Bullet objects in the instance variables which can not be deep copied. In this case, we actually want a shallow copy.
If you run the main branch with debug=True, you will see a lot of error messages saying bullet objects can not be deep copied. I already fixed it in the #310
